### PR TITLE
Developer docs updation

### PR DESCRIPTION
**Pull request by Ajayneethikannan**
This pull request attempts to solve the issue #3367 
The changes done are:
conversion of var to let and const,
conversion of normal functions to arrow functions wherever required.

The files which **do not** need updation are:

custom_p5_build.md
design_principles.md
issue_labels.md
preparing_a_pull_request.md
release_process.md
supported_browsers.md
webgl_mode_architecture.md

Some of my observations were:

The last code snippet in the friendly_error_system.md (which contains the attempt to access the global errors in the browser) produces infinite recursive calls and **stack overflow**, 
so maybe that issue can be corrected in a separate pull request . Please correct me if I am wrong

### CODE REVIEW

**Comment by limzykenneth** [comment]
It indeed does cause a stack overflow as objects are not copied when assigned to variables.

**Comment by Ajayneethikannan** [comment]
@limzykenneth should I remove that code snippet? I did not update it as it was given as an example of the attempted methods to catch the errors globally.

**Comment by limzykenneth** [comment]
It will be best if you can fix it instead of removing it, so probably saving individual references to `console.log` `console.warn` and `console.error` instead of just the `console` object.

**Comment by Ajayneethikannan** [comment]
@limzykenneth I have fixed the snippet as per your advice, thank you!
I have added few comments to explain what is happening in that piece of code.

Also , we can remove the red colored  error box from the console by returning true in the 
**window.onerror** method. Maybe we can wrap it up in some function  as in this snippet or use callbacks to handle the error after that. If someone could guide me, I would love to take it up as a new issue !

File: developer_docs/friendly_error_system.md [hunk 1]
```diff
@@ -116,10 +116,18 @@ line(0, 0, 100, 100, x3, Math.PI);
 * Global Error catching. It would be very helpful to catch the errors the browser is throwing to the console, so we can match them up with friendly comments. So far we've tried `window.onerror` and the following with no success.
```
Comment (lmccart) [inline developer_docs/friendly_error_system.md]: Is this snippet below something that works or does not work? In the previous version of this text, this paragraph was explaining something that had been tried but does not work.

Comment (lmccart) [reply developer_docs/friendly_error_system.md]: Please add 2 space indentations for this code block below.

Comment (Ajayneethikannan) [reply developer_docs/friendly_error_system.md]: @lmccart  I have done the indentation modification as directed. 

In the previous version of the code , 
```
var original = window.console;
      
["log", "warn", "error"].forEach(function(func){
window.console[func] = function(msg) {
    original[func].apply(original, arguments)
};
``` 

The newly defined function was referencing itself, causing the stack overflow.

This was corrected with help from @limzykenneth , 
making sure that the new functions do not reference themselves , by storing the references to the original function separately. 
```
const original = window.console;
const original_functions  = {
  log: original.log,
  warn:  original.warn,
  error: original.error
}

["log", "warn", "error"].forEach(function(func){
window.console[func] = function(msg) {
//do something with the msg caught by the wrapper function, then call the original function
original_functions[func].apply(original, arguments)
};
});
```

So the updated snippet does not cause the overflow and also catches the errors.

Also , I was not fully able to understand the use of catching global errors ,
but if we want  to replace the red error messages with better looking custom messages, 
we can replace the window.onerror method and finally return true , which may cause the code to run as if it did not detect the error, but also remove the 'unfriendly' error message :smile: 
Maybe to handle errors , we can wrap the window.onerror method with some other function, which returns true.

Thank you!

Comment (lmccart) [reply developer_docs/friendly_error_system.md]: Got it, thank you @Ajayneethikannan. Then can we remove this line above: 
"So far weve tried `window.onerror` and the following with no success."

And update this comment:
```js
//this snippet attempts to wrap window.console methods with a new function to modify their functionality"
```
to

```js
// this snippet wraps window.console methods with a new function to modify their functionality
// it is not currently implemented, but could be to give nicer formatted error messages
```

Then we should be good to merge. Thanks!

**Comment by lmccart** [comment]
🎉

File: developer_docs/friendly_error_system.md (hunk 1 above)
Comment (Ajayneethikannan) [reply developer_docs/friendly_error_system.md]: I have done the mentioned changes ,  please check it ,
Thank you @lmccart  !

**Pull request merged**